modules/cnf-provisioning-real-time-and-low-latency-workloads.adoc

```diff
@@ -0,0 +1,244 @@
+. (Optional) Add a node to the OCP cluster.
+See https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux_for_real_time/8/html-single/tuning_guide/index#Setting_BIOS_parameters[Setting BIOS parameters].
+. (Optional) Create a new MachineConfigPool for RT nodes.
+. Add the node to the proper MachineConfigPool, using node role labels.
++
+You need to decide which nodes will be configured with real-time workloads.
+It could be all of the nodes in the cluster or a subset of the nodes.
+The Performance Addon Operator expects all of the nodes are part of a dedicated MachineConfigPool.
+If you use all of the nodes, you just point the Performance Addon Operator to the worker node role label.
+If you use a subset, you need to group the nodes into a new MachineConfigPool.
+
+. Create the PerformanceProfile with the proper set of housekeeping cores and `realTimeKernel: enabled: true`.
+
+. Specify a node selector in the PerformanceProfile, as shown here:
++
+[source,yaml]
+----
+apiVersion: performance.openshift.io/v1
+kind: PerformanceProfile
+metadata:
+ name: example-performanceprofile
+spec:
+ [...]
+  realTimeKernel:
+    enabled: true
+  nodeSelector:
+   node-role.kubernetes.io/worker-rt: ""
+----
+
+. Verify that a matching MachineConfigPool exists with a label:
++
+----
+machineconfiguration.openshift.io/role=worker-rt
+----
+
+. OpenShift will start configuring the nodes, which may involve multiple reboots. Wait for the nodes to settle.
+This can take a long time depending on the specific hardware you use. We observed total time of about 20 minutes per node.
+
+. Verify everything is working as expected.
+
+== Verifying the real-time kernel was installed
+
+Use this command to verify that the real-time kernel is installed:
+
+----
+oc get node -o wide
+----
+
+Note the worker with the role `worker-rt` that contains the string `4.18.0-211.rt5.23.el8.x86_64`:
+----
+NAME                               	STATUS   ROLES           	AGE 	VERSION                  	INTERNAL-IP
+EXTERNAL-IP   OS-IMAGE                                       	KERNEL-VERSION
+CONTAINER-RUNTIME
+cnf-worker-0.example.com	          Ready	 worker,worker-rt   5d17h   v1.19.0-rc.2+aaf4ce1-dirty
+128.66.135.107   <none>    	        Red Hat Enterprise Linux CoreOS 46.82.202008252340-0 (Ootpa)
+4.18.0-211.rt5.23.el8.x86_64   cri-o://1.19.0-90.rhaos4.6.git4a0ac05.el8-rc.1
+[...]
+----
+
+== Creating a workload that works in real-time
+
+Use the following procedures for preparing a workload that will use real-time capabilities.
+
+* Create a Pod with a QoS class of Guaranteed
+* (Optional) Disable CPU load balancing for DPDK
+* Assign a proper node selector
+
+When writing your applications, follow the general recommendations described here:
```

Comment:
described in link:https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux_for_real_time/8/html-single/tuning_guide/index#chap-Application_Tuning_and_Deployment[Application tuning and deployment].